src/transformers/pipelines/image2text_generation.py

```diff
@@ -0,0 +1,124 @@
+import enum
+from typing import List, Union
+
+from ..utils import (
+    add_end_docstrings,
+    is_flax_available,
+    is_tf_available,
+    is_torch_available,
+    is_vision_available,
+    logging,
+    requires_backends,
+)
+from .base import PIPELINE_INIT_ARGS, Pipeline
+
+
+if is_vision_available():
+    from PIL import Image
+
+    from ..image_utils import load_image
+
+if is_flax_available():
+    from ..models.auto.modeling_flax_auto import FLAX_MODEL_FOR_VISION_2_SEQ_MAPPING
+
+if is_tf_available():
+    from ..models.auto.modeling_tf_auto import TF_MODEL_FOR_VISION_2_SEQ_MAPPING
+
+if is_torch_available():
+    from ..models.auto.modeling_auto import MODEL_FOR_VISION_2_SEQ_MAPPING
+
+logger = logging.get_logger(__name__)
+
+
+class ReturnType(enum.Enum):
+    TENSORS = 0
+    TEXT = 1
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class Image2TextGenerationPipeline(Pipeline):
+    """
+    Image2Text Generation pipeline using a `AutoModelForVision2Seq`. This pipeline predicts a caption for a given
+    image.
+
+    This image to text generation pipeline can currently be loaded from pipeline() using the following task identifier:
+    "image2text-generation".
+
+    See the list of available models on
+    [huggingface.co/models](https://huggingface.co/models?pipeline_tag=image-to-text).
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        requires_backends(self, "vision")
+        if self.framework == "flax":
+            self.check_model_type(FLAX_MODEL_FOR_VISION_2_SEQ_MAPPING)
+        elif self.framework == "tf":
+            self.check_model_type(TF_MODEL_FOR_VISION_2_SEQ_MAPPING)
+        else:
+            self.check_model_type(MODEL_FOR_VISION_2_SEQ_MAPPING)
+
+    def _sanitize_parameters(
+        self, return_tensors=None, return_text=None, return_type=None, clean_up_tokenization_spaces=None, **kwargs
+    ):
+        preprocess_params, forward_params, postprocess_params = {}, {}, {}
+
+        if return_tensors is not None and return_type is None:
+            return_type = ReturnType.TENSORS if return_tensors else ReturnType.TEXT
+        if return_type is not None:
+            postprocess_params["return_type"] = return_type
+
+        if clean_up_tokenization_spaces is not None:
+            postprocess_params["clean_up_tokenization_spaces"] = clean_up_tokenization_spaces
+
```

Comment:
I think these come from `text-generation` pipeline.

I am not sure we should enable them before it's necessary, returning tensors is not really aligned with the goal to focus on non ML practictionners.

I think returning pure text is better as a first pass (if users ask for more parameters, it's always easier to add later than be encumbered by the weight of being forced to support some parameters, and most importantly the parameter matrix that arises fast).

Reply:
Ok I simplified the generation.